src/main/java/com/example/hop_oasis/service/data/AuthenticationService.java

```diff
@@ -0,0 +1,71 @@
+package com.example.hop_oasis.service.data;
+
+import com.example.hop_oasis.dto.TokenResponse;
+import com.example.hop_oasis.dto.UserLoginRequest;
+import com.example.hop_oasis.dto.UserRegisterRequest;
+import com.example.hop_oasis.enums.Role;
+import com.example.hop_oasis.model.User;
+import com.example.hop_oasis.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class AuthenticationService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final AuthenticationManager authenticationManager;
+    private final JwtService jwtService;
+    private final UserAuthenticated userAuthenticated;
+
+    public TokenResponse register(UserRegisterRequest registerRequest) {
+        User user = User.builder()
+                .firstName(registerRequest.getFirstName())
+                .lastName(registerRequest.getLastName())
+                .email(registerRequest.getEmail())
+                .password(passwordEncoder.encode(registerRequest.getPassword()))
+                .role(Role.USER)
+                .build();
+        user = userRepository.save(user);
+        String jwtToken = jwtService.generateToken(user);
+
+
+        return TokenResponse.builder()
+                .accessToken(jwtToken)
+                .build();
+
+    }
+
+    public TokenResponse authenticate(UserLoginRequest loginRequest) {
+        authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        loginRequest.getEmail(),
+                        loginRequest.getPassword()
+                )
+        );
+        User user = userRepository.findByEmail(loginRequest.getEmail()).orElseThrow(() ->
+                new RuntimeException("User not found"));
+        String token = jwtService.generateToken(user);
+        log.info("token: " + token);
+        return TokenResponse.builder()
+                .accessToken(token)
+                .build();
+
+    }
+
+    public TokenResponse refreshToken() {
```

Comment:
If you want to provide refresh token feature, you need to have something, that would act as a refresh token, 
Smth that can verify that is the user that is trying to refresh its token)

Here, if attacker obtains a token, even expired, it can simply trigger /refresh/token and gain access to the system)